script/nspicker.js

```diff
@@ -0,0 +1,295 @@
+        }
+        e.stopPropagation();
+        e.preventDefault();
+        bc_nspicker.resultClick(this);
+        return false;
+    },
+
+    /**
+     * Handles the "click" on a given result anchor
+     */
+    resultClick: function(a){
+        //bc_nspicker.$entry.val(a.title);
+        //if(a.title == '' || a.title.substr(a.title.length-1) == ':'){
+        if(a.title == '' || a.title.substr(a.title.length-1) == ':'){
+            bc_nspicker.$entry.val(a.title);
+            bc_nspicker.autocomplete_exec();
+        }
+        else{
+            if (jQuery(a.nextSibling).is('span')) {
+                bc_nspicker.insertLink(a.nextSibling.innerHTML);
+            }else{
+                bc_nspicker.insertLink('');
+            }
+        }
+    },
+
+    /**
+     * Start the page/namespace lookup timer
+     *
+     * Calls autocomplete_exec when the timer runs out
+     */
+    autocomplete: function(){
+        if(bc_nspicker.timer !== null){
+            window.clearTimeout(bc_nspicker.timer);
+            bc_nspicker.timer = null;
+        }
+
+        bc_nspicker.timer = window.setTimeout(bc_nspicker.autocomplete_exec,350);
+    },
+
+    /**
+     * Executes the AJAX call for the page/namespace lookup
+     */
+    autocomplete_exec: function(){
+        var $res = jQuery(bc_nspicker.result);
+        bc_nspicker.deselect();
+        $res.html('<img src="'+DOKU_BASE+'lib/images/throbber.gif" alt="" width="16" height="16" />')
+            .load(
+            DOKU_BASE + 'lib/exe/ajax.php',
+            {
+                call: 'linkwiz',
+                q: bc_nspicker.$entry.val()
+            }
+        );
+    },
+
+    /**
+     * Show the link wizard
+     */
+    show: function(){
+        bc_nspicker.selection  = DWgetSelection(bc_nspicker.textArea);
+        bc_nspicker.$picker.show();
+        bc_nspicker.$entry.focus();
+        bc_nspicker.autocomplete();
+
+        // Move the cursor to the end of the input
+        var temp = bc_nspicker.$entry.val();
+        bc_nspicker.$entry.val('');
+        bc_nspicker.$entry.val(temp);
+    },
+
+    /**
+     * Hide the link wizard
+     */
+    hide: function(){
+        bc_nspicker.$picker.hide();
+        bc_nspicker.textArea.focus();
+    },
+
+    /**
+     * Toggle the link wizard
+     */
+    toggle: function(){
+        if(bc_nspicker.$picker.css('display') == 'none'){
+            bc_nspicker.show();
+        }else{
+            bc_nspicker.hide();
+        }
+    },
+
+    /**
+     * Select Namespace.
+     * Add all pages in the selected Namespace to the book, show
+     * window with added pages and then close/hide the Namespace
+     * picker.
+     */
+    selectNamespace: function(){
+        var $res = jQuery('#bc__nspicker_result a');
+        var $max = $res.length;
+        var $index;
+
+        $content = LANG.plugins.bookcreator.added_pages+"\n\n";
+
+        // Intentionally skip index 0
+        for ($index = 1 ; $index < $max ; $index++) {
```

Comment:
$res (may I propose `$pagelinks`/`$resultlinks` or whatever is a little more descriptive. (a personal preference ..)

$res is jquery object with an array of elements. This means you could just use https://api.jquery.com/each/ 
```js
$res.each(function(){ 
    ...
})
```